include/sys/endian.h

```diff
@@ -52,15 +52,15 @@
 /* Common byte swapping macros */
 
 #ifdef CONFIG_HAVE_BUILTIN_BSWAP16
-#  define __swap_uint16 __builtin_bswap16
+#  define __swap_uint16(n) ((uint16_t)__builtin_bswap16(n))
```

Comment:
I think that automatic cast is applied to function parameter. Why do we need an explicit cast?